pkg/gofr/datasource/mongo/mongo.go

```diff
@@ -133,94 +142,129 @@ func (c *Client) Find(ctx context.Context, collection string, filter, results in
+
+	defer c.pushObservability(&QueryLog{Query: "countDocuments", Collection: collection, Filter: filter}, time.Now(),
+		"countDocuments", span)
 
-	return c.Database.Collection(collection).CountDocuments(ctx, filter)
+	return result, err
 }
 
 // DeleteOne deletes a single document from the specified collection based on the provided filter.
 func (c *Client) DeleteOne(ctx context.Context, collection string, filter interface{}) (int64, error) {
-	defer c.sendOperationStats(&QueryLog{Query: "deleteOne", Collection: collection, Filter: filter}, time.Now())
+	tracedCtx, span := c.addTrace(ctx, "deleteOne", collection)
 
-	res, err := c.Database.Collection(collection).DeleteOne(ctx, filter)
+	res, err := c.Database.Collection(collection).DeleteOne(tracedCtx, filter)
 	if err != nil {
 		return 0, err
 	}
 
+	defer c.pushObservability(&QueryLog{Query: "deleteOne", Collection: collection, Filter: filter}, time.Now(),
+		"deleteOne", span)
+
 	return res.DeletedCount, nil
 }
 
 // DeleteMany deletes multiple documents from the specified collection based on the provided filter.
 func (c *Client) DeleteMany(ctx context.Context, collection string, filter interface{}) (int64, error) {
-	defer c.sendOperationStats(&QueryLog{Query: "deleteMany", Collection: collection, Filter: filter}, time.Now())
+	tracedCtx, span := c.addTrace(ctx, "deleteMany", collection)
 
-	res, err := c.Database.Collection(collection).DeleteMany(ctx, filter)
+	res, err := c.Database.Collection(collection).DeleteMany(tracedCtx, filter)
 	if err != nil {
 		return 0, err
 	}
 
+	defer c.pushObservability(&QueryLog{Query: "deleteMany", Collection: collection, Filter: filter}, time.Now(),
+		"deleteMany", span)
+
 	return res.DeletedCount, nil
 }
 
 // Drop drops the specified collection from the database.
 func (c *Client) Drop(ctx context.Context, collection string) error {
-	defer c.sendOperationStats(&QueryLog{Query: "drop", Collection: collection}, time.Now())
+	tracedCtx, span := c.addTrace(ctx, "drop", collection)
+
+	err := c.Database.Collection(collection).Drop(tracedCtx)
 
-	return c.Database.Collection(collection).Drop(ctx)
+	defer c.pushObservability(&QueryLog{Query: "drop", Collection: collection}, time.Now(), "drop", span)
+
+	return err
 }
 
 // CreateCollection creates the specified collection in the database.
 func (c *Client) CreateCollection(ctx context.Context, name string) error {
-	defer c.sendOperationStats(&QueryLog{Query: "createCollection", Collection: name}, time.Now())
+	tracedCtx, span := c.addTrace(ctx, "createCollection", name)
+
+	err := c.Database.CreateCollection(tracedCtx, name)
+
+	defer c.pushObservability(&QueryLog{Query: "createCollection", Collection: name}, time.Now(), "createCollection",
+		span)
 
-	return c.Database.CreateCollection(ctx, name)
+	return err
 }
 
-func (c *Client) sendOperationStats(ql *QueryLog, startTime time.Time) {
+func (c *Client) pushObservability(ql *QueryLog, startTime time.Time, method string, span trace.Span) {
```

Comment:
this method name is a convention followed by GoFr and being used everywhere. If renaming the name, please take note that we would need to update this everywhere later...

Reply:
renamed back to `sendOperationStats `